pkg/mesh/mesh.go

```diff
@@ -700,18 +686,22 @@ func peersAreEqual(a, b *Peer) bool {
 	}
 	// Check the DNS name first since this package
 	// is doing the DNS resolution.
-	if !a.Endpoint.Equal(b.Endpoint, true) {
+	if a.Endpoint.StringOpt(false) != b.Endpoint.StringOpt(false) {
```

Comment:
```suggestion
	if a.Endpoint.StringOpt(true) != b.Endpoint.StringOpt(true) {
```

Reply:
Hmm I'm not familiar with this lib

Reply:
Maybe it is not a good idea to compare strings here. Anyways the comment above suggests to check DNS names first, so this is the boolean flag about. Maybe `wireguard.Endpoint` should also implement `Equal(dnsFirst)`?

Reply:
It used to implement this, right? So you mean un-remove the functionality? I think that could be helpful

Reply:
We are using it in nodesAreEqual and peersAreEqual.
I think comparing the strings is save. So the Equal func could just call the String method.
Is this a good trade off for adding another function to wireguard.Endpoint?

Reply:
yeah, I think it's worth the complexity to make it easier for the callers so that they don't need to remember to check the DNS equality before the IP equality